src/HealthChecks.Uris/UriHealthCheck.cs

```diff
@@ -57,6 +57,15 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                             return new HealthCheckResult(context.Registration.FailureStatus, description: $"Discover endpoint #{idx} is not responding with code in {expectedStatusCodes.Min}...{expectedStatusCodes.Max} range, the current status is {response.StatusCode}.");
                         }
 
+                        if (!string.IsNullOrWhiteSpace(item.ExpectedContent))

```

Comment:
```suggestion
                        if (item.ExpectedContent != null)
```

Reply:
could you ellaborate on this? Why is this in improvement?

Reply:
Because content _may_ be empty and because empty content _may_ be "good" content. For exmple, endpont may return some content only in case of error conditions. For other cases (normal work) it just returns nothing (empty body).

Reply:
never thought of that, nice!